src/libraries/System.IO.Pipelines/src/PACKAGE.md

```diff
@@ -2,43 +2,113 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+A library designed to make it easier to do high-performance I/O.
 
+Apps that parse streaming data are composed of boilerplate code having many specialized and unusual code flows.
+The boilerplate and special case code is complex and difficult to maintain.
+
+`System.IO.Pipelines` was architected to:
+
+* Have high performance parsing streaming data.
+* Reduce code complexity.
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Single producer/single consumer byte buffer management.
+* Reduction in code complexity and boilerplate code associated with I/O operations.
 
 ## How to Use
 
 <!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->
 
+The `Pipe` class can be used to create a `PipeWriter`/`PipeReader` pair.
+All data written into the `PipeWriter` is available in the `PipeReader`.
+
+```csharp
```

Comment:
Actually, can we please remove the sample code from here and instead just link to the docs, which have samples already? Trying to keep the number of samples in different places down.